src/graph.py

```diff
@@ -86,3 +86,24 @@ def adjacent(self, val1, val2):
         if val2 in self[val1]:
             return True
         return False
+
+    def depth_first_traversal(self, start_val, path=[]):
+        """Return the path with depth transversal."""
+        if start_val not in self.keys():
+            raise ValueError('No such starting value')
+        if start_val not in path:
+            path.append(start_val)
+            for val in self.neighbors(start_val):
+                self.depth_first_traversal(val, path)
+        return path
+
+    def breadth_first_traversal(self, start_val):
+        """Return the path with breadth transversal."""
+        if start_val not in self.keys():
+            raise ValueError('No such starting value')
+        path = [start_val]
+        for node in path:
+            for val in self[node]:
+                if val not in path:
+                    path.append(val)
```

Comment:
What's the look up time for this check?  Consider the trade off of time/space in using an alternate data structure to store 'visited' nodes (but not the path - must be a list as it is here).